tools/eval/datasets/pp_humanseg.py

```diff
@@ -0,0 +1,209 @@
+        self.miou = -1
+        self.class_miou = -1
+        self.acc = -1
+        self.class_acc = -1
+
+
+    @property
+    def name(self):
+        return self.__class__.__name__
+    
+
+    def load_data(self, val_path) : 
+        """
+        Load validation image set from val.txt file
+        Args :
+            val_path (str) : path to val.txt file
+        Returns :
+            image_set (list) : list of image path of input and expected image
+        """
+
+        image_set = []
+        with open(val_path, 'r') as f : 
+            for line in f.readlines() : 
+                image_set.append(line.strip().split())
+
+        return image_set
+    
+    
+    def eval(self, model) : 
+        """
+        Evaluate model on validation set
+        Args :
+            model (object) : PP_HumanSeg model object
+        """
+
+        pbar = tqdm(self.image_set)
+
+        for input_image, expected_image in pbar : 
+            pbar.set_description(
+                "Evaluating {} with {} val set".format(model.name, self.name))
+            
+            input_image = cv.imread(os.path.join(self.root, input_image))
+            input_image = cv.resize(input_image, (192, 192))
+
+            expected_image = cv.imread(os.path.join(self.root, expected_image),cv.IMREAD_GRAYSCALE)
+            expected_image = cv.resize(expected_image, (192, 192))[np.newaxis, :, :]
+
+
+            output_image = model.infer(input_image)
+            
+            intersect_area_all = []
+            pred_area_all = []
+            label_area_all = []
+
+
+            intersect_area, pred_area, label_area = self.calculate_area(
+                output_image,
+                expected_image,
+                self.num_classes)
+            intersect_area_all = intersect_area_all + intersect_area
+            pred_area_all = pred_area_all + pred_area
+            label_area_all = label_area_all + label_area
+
+
+            
+        self.class_iou, self.miou = self.mean_iou(intersect_area_all, pred_area_all,
+                                           label_area_all)
+        self.class_acc, self.acc = self.accuracy(intersect_area_all, pred_area_all)
+    
+
+    def get_results(self) :
+        """
+        Get evaluation results
+        Returns :
+            miou (float) : mean iou
+            class_miou (list) : iou on all classes
+            acc (float) : mean accuracy
+            class_acc (list) : accuracy on all classes
+        """
+        return self.miou, self.class_miou, self.acc, self.class_acc
+    
+
+    def print_result(self) : 
+        """
+        Print evaluation results
+        """
+        print("Mean IoU : ", self.miou)
+        print("Class IoU : ", self.class_iou)
+        print("Mean Accuracy : ", self.acc)
+        print("Class Accuracy : ", self.class_acc)
+
+    
+    def one_hot(self, arr, max_size) :
+        return np.eye(max_size)[arr]
+
+
+    def calculate_area(self,pred, label, num_classes, ignore_index=255):
```

Comment:
Could you provide some test data for validating the correctness of this function?

Reply:
For the following test : 
``` 
# tensor of size (1,x,x)
pred = np.array([
    [   [0, 0, 0, 0, 0],
        [0, 0, 1, 1, 0],
        [0, 0, 1, 1, 0],
        [0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0]],
])
label = np.array([
    [   [0, 0, 0, 0, 0],
        [0, 0, 1, 1, 0],
        [0, 0, 1, 1, 0],
        [0, 0, 0, 1, 0],
        [0, 0, 0, 0, 0]],
])

area = calculate_area(pred, label, 2)
print(f"intersect_area: {area[0]}  pred_area: {area[1]}  label_area: {area[2]}")
iou = mean_iou(*area)
print(f"IOU on classes: {iou[0]}  mean IOU: {iou[1]}")
```

Here's the output : 
![image](https://user-images.githubusercontent.com/94868003/218070001-7fe9b154-85bd-4469-822c-6d03060c5925.png)

It can be seen from the pred and label tensor that intersect_area of 0s is 20 and intersect area of 1s is 4.

For IOU : 
union for class 0 : pred + label - interset = 20 + 21 - 20 = 21
iou for class 0 : intersect/union = 20/21 = 0.9523
union for class 1 : pred + label - interset  4 + 5 - 4 = 5
iou for class 1 : intersect/union = 4/5 = 0.8